compiler/next/include/chpl/util/hash.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_UTIL_HASH_H
+#define CHPL_UTIL_HASH_H
+
+#include <cstddef>
+#include <functional>
+#include <string>
+#include <tuple>
+#include <utility>
+
+namespace chpl {
+
+// Combine two hash functions
+inline size_t hash_combine(size_t hash, size_t other) {
+  return hash ^ (other + 0x9e3779b9 + (hash << 6) + (hash >> 2));
+}
+
+// FNV-1a hash function
```

Comment:
You are welcome to leave the FNV-1a hash function here but I earlier requested that you compare FNV-1a with the hash function we had. (Or, of course, find another comparison, but I suspect it will be easier to to our own short one).

Specifically, compare the number of collisions when taking `hash & size` for a variety of plausible sizes for hashtables when working with keys from some realistic scenario. Compare also the runtime.

It appears that this PR changes to FNV-1a without doing such a comparison.